src/view/analytics/seen.js

```diff
@@ -46,7 +8,8 @@ const getObserver = handleEvent => {
 export const manageSeenEvents = prompts => {
 	prompts.forEach( prompt => {
 		const handleEvent = () => {
-			const payload = getEventPayload( 'seen', getRawId( prompt.getAttribute( 'id' ) ) );
+			const promptId = getRawId( prompt.getAttribute( 'id' ) );
+			const payload = getEventPayload( 'seen', promptId );
 			sendEvent( payload );
 		};
```

Comment:
Non-blocking nit: dispatching the `prompt_seen` activity triggers an event that can get picked up through `ras.on( 'activity', cb )`. It might be better to use that API for the same goal instead of running another intersection observer.

Reply:
The logic is backwards, though—the intersection observer is what triggers the `prompt_seen` activity, not the other way around. We only want to dispatch `prompt_seen` once the prompt is visible in the viewport for a certain period of time.

Reply:
Is it not like that in the code? My suggestion is to make the analytics read from `prompt_seen` event trigger ([which is already behind an intersection observer](https://github.com/Automattic/newspack-popups/blob/7e1a92b0dd26850f92a695da12f6aa540c798aa2/src/view/utils/index.js#L68-L78)) instead of having another intersection observer.

Reply:
Oh! I see what you mean. You're right, we should trigger the analytics event based on the activity dispatch.

Reply:
173e03d refactors the `seen` GA4 event so it gets triggered based on `ras.on( 'activity' )` instead of its own intersection observer.